addons/binding/org.openhab.binding.nibeuplink/utils/create_channel_types.awk

```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/awk -f
+
+function ltrim(s) { sub(/^[ \t\r\n]+/, "", s); return s }
+function rtrim(s) { sub(/[ \t\r\n]+$/, "", s); return s }
+function trim(s) { return rtrim(ltrim(s)); }
+
+BEGIN{
+	FS=";"
+	print "<?xml version=\"1.0\" encoding=\"UTF-8\"?>"
+	print "<thing:thing-descriptions bindingId=\"nibeuplink\""
+	print "xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" "
+	print "xmlns:thing=\"http://eclipse.org/smarthome/schemas/thing-description/v1.0.0\" "
+	print "xsi:schemaLocation=\"http://eclipse.org/smarthome/schemas/thing-description/v1.0.0 http://eclipse.org/smarthome/schemas/thing-description-1.0.0.xsd\">"
+	print ""
+}
+NR>5{
+
+	title=$1
+	info=$2
+	id=$3
+	unit=$4
+	size=$5
+	factor=$6
+	min=$7
+	max=$8
+	idefault=$9
+	mode=$10
+
+	gsub("\"", "", title)
+	gsub("\"", "", info)
+	gsub("\"", "", unit)
+
+	optionsData=info
+	delete keys
+	delete values
+	
+	# parse options from info field
+	if (index(optionsData, "=") > 0)
+	{
+		gsub(",", "=", optionsData)
+		gsub(" ", "=", optionsData)
+		
+		key=""
+		value=""
+		keyCount=1
+		
+		count = split(optionsData, t, "=")
+		
+		for (i = 0; ++i <= count;)
+		{
+			if (t[i] ~ /^[0-9]+$/)
+			{
+				if (key != "")
+				{
+					if (trim(value) == "")
+					{
+						values[keyCount-1] = values[keyCount-1] " " key
+					}
+					else
+					{
+						keys[keyCount] = trim(key)
+						values[keyCount] = trim(value)
+						keyCount++
+						key=""
+						value=""
+					}
+				}
+				
+				key = t[i]
+			}
+			else
+			{
+				if (key != "")
+					value = value " " t[i]
+			}
+		}
+		
+		if (key != "")
+		{
+			keys[keyCount] = trim(key)
+			values[keyCount] = trim(value)
+		}
+	}
+
+
+	printf("	<channel-type id=\"%s-type-%s\">\n", model, id)
+
+	if (min == 0 && max == 1)
+	{
+		printf("		<item-type>Switch</item-type>\n")
+		printf("		<label>%s</label>\n", title)
+		printf("		<description>%s</description>\n", info)
+		
+		if (mode == "R")
+			readOnly="true"
+		else 
+			readOnly="false"
+
+		printf("		<state readOnly=\"%s\"></state>\n", readOnly)
+
+	}
+	else
+	{
+		if (keys[1] != "")
+			printf("		<item-type>String</item-type>\n")
+		else
+			printf("		<item-type>Number</item-type>\n")
+			
+		printf("		<label>%s</label>\n", title)
+		printf("		<description>%s</description>\n", info)
+		
+		if (min == 0 && max == 0) 
+		{
+			min="0";
+			if (size == "u8")
+				max="255"
+			if (size == "u16")
+				max="65535"
+			if (size == "u32")
+				max="4294967295"
+
+			if (size == "s8")
+			{
+				min="-128"
+				max="127"
+			}
+			if (size == "s16")
+			{
+				min="-32767"
```

Comment:
Is the value -32767 correct? Looking at the other negative numbers I would expect this value to be: -32768?

Reply:
I don't know I retrieved these scripts from another user and made only little adoptions. But anyway this should not have any effect on the code. It just generates the MD file.